src/worker.js

```diff
@@ -83,42 +83,48 @@ function fixJob({ cliEngineOptions, contents, eslint, filePath }) {
 
 module.exports = async () => {
   process.on('message', (jobConfig) => {
-    const {
-      contents, type, config, filePath, projectPath, rules, emitKey
-    } = jobConfig
-    if (config.disableFSCache) {
-      FindCache.clear()
-    }
-
-    const fileDir = Path.dirname(filePath)
-    const eslint = Helpers.getESLintInstance(fileDir, config, projectPath)
-    const configPath = Helpers.getConfigPath(fileDir)
-    const noProjectConfig = (configPath === null || isConfigAtHomeRoot(configPath))
-    if (noProjectConfig && config.disableWhenNoEslintConfig) {
-      emit(emitKey, { messages: [] })
-      return
-    }
-
-    const relativeFilePath = Helpers.getRelativePath(fileDir, filePath, config, projectPath)
-
-    const cliEngineOptions = Helpers
-      .getCLIEngineOptions(type, config, rules, relativeFilePath, fileDir, configPath)
+    // We catch all worker errors so that we can create a separate error emitter
+    // for each emitKey, rather than adding multiple listeners for `task:error`
+    try {
+      const {
```

Comment:
Minor nit, but I'd leave this out of the `try/catch` and just use `emitKey` in the `catch` block.